HHS/Head-Start-TTADP
# Update goal status based on objective status

## Description of change

Use sequelize's "afterCreate" and "afterUpdate" hooks to automatically move a goal through the statuses based [on this logical diagram](https://app.mural.co/t/adhoccorporateworkspace2583/m/adhoccorporateworkspace2583/1647548115774/d89d9477fdc3555b8b39b382311d8f2d1f2bacfc?sender=u775286a3a84f5c5204174211).

This is necessary for MVP since the only way to change the goal status in the UI is via the RTR.

## How to test
Verify that the following statements are true:
- Create a goal on the Create Goals form by filling out the form and clicking Save Draft but DO NOT submit. The goal status should be "Draft"
- Make sure all the objectives are "Not Started" and click submit. The status should then be "Not Started"
- Repeat the steps above, but set one objective to "In Progress." The goal status should be "In Progress."
- On an activity report, a new goal should be "Draft" until the report is approved. Then, it becomes "In Progress."
- A "Not Started" goal created on the Create Goals form, if used on an Activity Report, should be moved to "In Progress" regardless of objective status.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1004

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
